docs/tutorials/particle-filtering.md

```diff
@@ -82,43 +78,49 @@ int main() {
-        ranges::views::remove_if([&](double range) { return std::abs(range) > parameters.sensor_range; }) |  //
+    const auto range_measurements =
+        parameters.landmark_map |                                                                                   //
+        ranges::views::transform([&](double landmark_position) { return landmark_position - current_position; }) |  //
+        ranges::views::remove_if([&](double range) { return std::abs(range) > parameters.sensor_range; }) |         //
         ranges::to<std::vector>;
 
-    auto sensor_model = [&](double particle_position) {
-      auto particle_detections =
-          parameters.landmark_map |                                                                  //
-          ranges::views::transform([&](double landmark) { return landmark - particle_position; }) |  //
+    const auto sensor_model = [&](double position) {
+      auto range_map =
+          parameters.landmark_map |                                                                           //
+          ranges::views::transform([&](double landmark_position) { return landmark_position - position; }) |  //
           ranges::to<std::vector>;
 
       return parameters.min_particle_weight +
              std::transform_reduce(
-                 detections.begin(), detections.end(), 1.0, std::multiplies<>{}, [&](double detection) {
-                   auto distances =           //
-                       particle_detections |  //
-                       ranges::views::transform(
-                           [&](double particle_detection) { return std::abs(detection - particle_detection); });
+                 range_measurements.begin(), range_measurements.end(), 1.0, std::multiplies<>{},
+                 [&](double range_measurement) {
+                   const auto distances = range_map | ranges::views::transform([&](double range) {
+                                            return std::abs(range - range_measurement);
+                                          });
                    const auto min_distance = ranges::min(distances);
                    return std::exp((-1 * std::pow(min_distance, 2)) / (2 * parameters.sensor_model_sigma));
                  });
     };
 
+    record.current = particles;
+
     particles |= beluga::actions::propagate(std::execution::seq, motion_model);
+    record.propagate = particles;
 
     particles |= beluga::actions::reweight(std::execution::seq, sensor_model) | beluga::actions::normalize;
+    record.reweight = particles;
 
     particles |= beluga::views::sample |                                        //
                  ranges::views::take_exactly(parameters.number_of_particles) |  //
                  beluga::actions::assign;
+    record.resample = particles;
 
     const auto estimation = beluga::estimate(beluga::views::states(particles), beluga::views::weights(particles));
   }
```

Comment:
delete all the code related to records in this tutorial

```suggestion
    };

    particles |= beluga::actions::propagate(std::execution::seq, motion_model);

    particles |= beluga::actions::reweight(std::execution::seq, sensor_model) | beluga::actions::normalize;

    particles |= beluga::views::sample |                                        //
                 ranges::views::take_exactly(parameters.number_of_particles) |  //
                 beluga::actions::assign;

    const auto estimation = beluga::estimate(beluga::views::states(particles), beluga::views::weights(particles));
  }
```

Reply:
Ahh, good catch. I thought I had removed it but clearly didn't.